src/nvim/ex_docmd.c

```diff
@@ -4732,7 +4732,7 @@ bool before_quit_autocmds(win_T *wp, bool quit_all, bool forceit)
 /// ":{nr}quit": quit window {nr}
 static void ex_quit(exarg_T *eap)
 {
-  if (cmdwin_type != 0) {
+  if (curwin == cmdwin_win) {
```

Comment:
Can use this to close other windows when in the messages window, which can allow for `cmdwin_old_curwin` to be closed or the cmdwin to be left as the only window; that can cause strangeness.

E.g: this causes E199, which shouldn't happen:

```vim
only!
lua require"vim._extui".enable{}
echom "foo"
call feedkeys("q:", "nt")
au CmdwinEnter * ++once mess | 1q! | qa! | au SafeState * ++once au SafeState * ++once mes
```